paimon-core/src/main/java/org/apache/paimon/table/source/snapshot/CreationTimestampStartingScanner.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.paimon.table.source.snapshot;
+
+import org.apache.paimon.CoreOptions;
+import org.apache.paimon.manifest.PartitionEntry;
+import org.apache.paimon.table.source.ScanMode;
+import org.apache.paimon.utils.ChangelogManager;
+import org.apache.paimon.utils.SnapshotManager;
+
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * {@link StartingScanner} for the {@link CoreOptions.StartupMode#FROM_CREATION_TIMESTAMP} startup
+ * mode.
+ */
+public class CreationTimestampStartingScanner extends AbstractStartingScanner {
+
+    private final AbstractStartingScanner scanner;
+
+    public CreationTimestampStartingScanner(
+            SnapshotManager snapshotManager,
+            ChangelogManager changelogManager,
+            long creationMillis,
+            boolean changelogDecoupled,
+            boolean isStreaming) {
+        super(snapshotManager);
+        Long startingSnapshotPrevId =
+                TimeTravelUtil.earlierThanTimeMills(
+                        snapshotManager,
+                        changelogManager,
+                        creationMillis,
+                        changelogDecoupled,
+                        true);
+        Optional.ofNullable(startingSnapshotPrevId)
+                .map(id -> id + 1)
```

Comment:
Why not just use `TimeTravelUtil.laterOrEaqual` so that we don't need to map `id + 1`.

Reply:
Here isn't later or equal, if the ts is earlier than the earliest snapshot or changelog, we need to get null.